search-provider/eks-discovery-feed-provider.c

```diff
@@ -230,6 +261,20 @@ article_card_descriptions_cb (GObject *source,
       add_key_value_pair_from_model_to_variant (model, &builder, "thumbnail-uri");
       add_key_value_pair_from_model_to_variant (model, &builder, "ekn-id");
 
+      /* Select a random clickbait title and add that */
+      g_autoptr(GVariant) clickbaityTitleVariant;
+      g_object_get (model, "clickbaity-titles", &clickbaityTitleVariant, NULL);
+      g_autoptr(GPtrArray) clickbaityTitlesArray = variant_array_string_to_ptr_array(clickbaityTitleVariant,
+                                                                                     ESTIMATED_CLICKBAITY_TITLES);
+
+      if (clickbaityTitlesArray->len)
+        {
+          add_key_value_pair_to_variant (&builder,
+                                         "clickbaity_title",
+                                         g_ptr_array_index(clickbaityTitlesArray,
+                                                           rand () % clickbaityTitlesArray->len));
```

Comment:
It feels wrong to go from variant to pointer array to variant again like this.
You can just use `g_variant_get_child()`, or `g_variant_get_child_value()` on the original variant instead.